### ISSUE TITLE
Add rule: prefer-nullish-coalescing

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#9]
## Problem

LLMs use `||` for defaults instead of `??`, incorrectly treating `0` and empty strings as falsy.

## Rule

Prefer `??` over `||` for default values.

### ❌ Bad
```ts
const name = input || "Unknown"; // Fails for "" and 0
const count = value || 10; // Fails for 0
```

### ✅ Good
```ts
const name = input ?? "Unknown";
const count = value ?? 10;
```

## Why it matters

- `||` treats 0, "", false as falsy
- `??` only treats null/undefined as nullish
- Prevents subtle bugs with valid falsy values
